connectors/src/connectors/github/index.ts

```diff
@@ -282,162 +291,158 @@ export class GithubConnectorManager extends BaseConnectorManager<null> {
+                connectorId: c.id,
+                parentInternalId,
+              },
+            }),
+          ]);
 
-        const repo = repoRes.value;
-
-        const nodes: ContentNode[] = [];
-
-        if (latestIssue) {
-          nodes.push({
-            provider: c.type,
-            internalId: `${repoId}-issues`,
-            parentInternalId,
-            type: "database",
-            title: "Issues",
-            sourceUrl: repo.url + "/issues",
-            expandable: false,
-            permission: "read",
-            dustDocumentId: null,
-            lastUpdatedAt: latestIssue.updatedAt.getTime(),
+          files.sort((a, b) => {
+            return a.fileName.localeCompare(b.fileName);
+          });
+          directories.sort((a, b) => {
+            return a.dirName.localeCompare(b.dirName);
           });
-        }
 
-        if (latestDiscussion) {
-          nodes.push({
-            provider: c.type,
-            internalId: `${repoId}-discussions`,
-            parentInternalId,
-            type: "channel",
-            title: "Discussions",
-            sourceUrl: repo.url + "/discussions",
-            expandable: false,
-            permission: "read",
-            dustDocumentId: null,
-            lastUpdatedAt: latestDiscussion.updatedAt.getTime(),
+          const nodes: ContentNode[] = [];
+
+          directories.forEach((directory) => {
+            nodes.push({
+              provider: c.type,
+              internalId: directory.internalId,
+              parentInternalId,
+              type: "folder",
+              title: directory.dirName,
+              sourceUrl: directory.sourceUrl,
+              expandable: true,
+              permission: "read",
+              dustDocumentId: null,
+              lastUpdatedAt: directory.codeUpdatedAt.getTime(),
+            });
           });
-        }
 
-        if (codeRepo) {
-          nodes.push({
-            provider: c.type,
-            internalId: `github-code-${repoId}`,
-            parentInternalId,
-            type: "folder",
-            title: "Code",
-            sourceUrl: repo.url,
-            expandable: true,
-            permission: "read",
-            dustDocumentId: null,
-            lastUpdatedAt: codeRepo.codeUpdatedAt.getTime(),
+          files.forEach((file) => {
+            nodes.push({
+              provider: c.type,
+              internalId: file.documentId,
+              parentInternalId,
+              type: "file",
+              title: file.fileName,
+              sourceUrl: file.sourceUrl,
+              expandable: false,
+              permission: "read",
+              dustDocumentId: file.documentId,
+              lastUpdatedAt: file.codeUpdatedAt.getTime(),
+            });
           });
-        }
 
-        return new Ok(nodes);
+          return new Ok(nodes);
+        }
+        // we should never be getting issues, discussions or code files as parent
+        case "REPO_ISSUES":
```

Comment:
repo issues and repo discussions seem legitimate as parents, right? they are parents of all issues and all discussions

Reply:
it seems that former logic also excluded them but I'm a bit surprised

Reply:
I think we don't ever need to retrieve the parents of a single issue or discussion, therefore we can exclude these 2 cases
That said I can add them